### PR TITLE
Felix/remove kinect

### PR DESCRIPTION
**Pull request by FelixKHWong**
Removing the kinect from the movo and adding in the realsense for ROS and for RVIZ

### CODE REVIEW

File: movo_common/movo_description/urdf/movo_components/movo_pan_tilt.urdf.xacro [hunk 1]
```diff
@@ -136,9 +136,10 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
             </inertial>
 		</link>
         
-		<xacro:kinect_one_sensor prefix="${rgbd_prefix}" parent="tilt_link">
-			<origin xyz="0.05825 -0.0105875 -0.026" rpy="${-M_PI/2} 0 ${-M_PI/2}" />
-		</xacro:kinect_one_sensor>
+		<xacro:sensor_d435 parent="tilt_link">
+			<origin xyz="0 0 0" rpy="${-M_PI/2} 0 ${-M_PI/2}" />
```
Comment (FelixKHWong) [inline movo_common/movo_description/urdf/movo_components/movo_pan_tilt.urdf.xacro]: These numbers may need changing

File: movo_desktop/movo_viz/rviz/robot_.rviz [hunk 2]
```diff
@@ -634,7 +639,7 @@ Visualization Manager:
       Value: true
     - Class: rviz/Image
       Enabled: false
-      Image Topic: /kinect2/sd/image_color_rect
+      Image Topic: /camera/depth/image_rect_raw
```
Comment (FelixKHWong) [inline movo_desktop/movo_viz/rviz/robot_.rviz:642]: might change this topic to be the color version but this name seemed closer to the previous

**Comment by younesr1** [comment]
have u run `roslaunch movo_viz view_robot.launch` to verify the camera shows up correctly? also what things have u done to try to validate this change? maybe fix movo2.launch first then run it and make sure these changes work as expected

**Comment by FelixKHWong** [comment]
Sure I'll give that a shot

**Review by bdschnapp — APPROVED** [review]
looks good to me, only hesitation is it seems like there were significantly more files removed than added. I'm assuming this is just the kinect and realsesne being structured differently?

**Comment by FelixKHWong** [comment]
I've added some changes so that the realsense is in the correct spot in the movo model and I've added some changes to be able to properly launch the rviz files. I solved the errrors with movo2.launch. The solution for the time sync issue is to add a printf with the password and pipe it to the sudo -S command. The password for my computer is not the robohub one so I won't be adding it but I'll put in the necessary change for the movo
realsense position

**Comment by FelixKHWong** [comment]
Please do check if the script issue with the password is feasible or not. I'm assuming since the password is that it is fine to put in in the script but we can remove it if needed

File: movo_desktop/movo_viz/launch/view_robot.launch [hunk 3]
```diff
@@ -3,6 +3,13 @@
 <launch>
   <arg name='function' default=''/>
 
+  <include file="$(find movo_description)/launch/movo_upload.launch"/>
+
+  <include file="$(find movo_moveit_config)/launch/move_group.launch">
```
Comment (younesr1) [inline movo_desktop/movo_viz/launch/view_robot.launch]: its a little strange for the view_robot launch file to be launching move groups. isnt the point of this launch file to just view the robot?

**Comment by younesr1** [comment]
> I've added some changes so that the realsense is in the correct spot in the movo model and I've added some changes to be able to properly launch the rviz files. I solved the errrors with movo2.launch. The solution for the time sync issue is to add a printf with the password and pipe it to the sudo -S command. The password for my computer is not the robohub one so I won't be adding it but I'll put in the necessary change for the movo realsense position

I dont think i understand what you mean. you've run movo2.launch locally? can u collect a rosbag of it running and post it? also writing passwords in source code is not great imo. there should be a better way

File: movo_desktop/movo_viz/launch/view_robot.launch (hunk 3 above)
Comment (FelixKHWong) [reply movo_desktop/movo_viz/launch/view_robot.launch]: Yeah that's what I thought but when I try to run the view_robot.launch file it raises an error about not having the movo description and after I add the upload launch file it states that I need the semantic description. Did you not see these when you ran it?

**Comment by FelixKHWong** [comment]
Yeah I was running movo2 launch locally. It doesn't detect the realsense so it probably doesn't do anything but it was raising the time sync error we saw before so I worked on fixing it. I can try to add the user to the list of sudoers but so far when I've tried it it hasn't worked

**Comment by FelixKHWong** [comment]
Removed password from history

**Comment by FelixKHWong** [comment]
[2022-10-11-13-20-35.zip](https://github.com/nightingale-project/kinova-movo/files/9758398/2022-10-11-13-20-35.zip)
Here's the rosbag file which I had to zip to put here. It seems liek tehres a movo group error and a python error regarding the planning scene interface. I'll take a look at those

**Comment by FelixKHWong** [comment]
https://github.com/ros-industrial/universal_robot/issues/413 https://github.com/ros-planning/moveit_tutorials/issues/513. Seems like a move_group was deprecated and changed

File: movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-import thread, copy
+import _thread, copy
```
Comment (FelixKHWong) [inline movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py]: library updated for python3

File: movo_moveit_config/launch/move_group.launch [hunk 5]
```diff
@@ -51,7 +51,7 @@
 
     <!-- MoveGroup capabilities to load -->
     <param name="capabilities" value="move_group/MoveGroupCartesianPathService
-				      move_group/MoveGroupExecuteService
+				      move_group/MoveGroupExecuteTrajectoryAction
```
Comment (FelixKHWong) [inline movo_moveit_config/launch/move_group.launch]: Old group was deprecated and replaced with new group

**Review by le2tang — APPROVED** [review]

File: movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py (hunk 4 above)
Comment (younesr1) [reply movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py]: is this right? have u seen this?
https://stackoverflow.com/questions/5568555/thread-vs-threading
"_thread is infrastructure code that is used to implement threading, and normal Python code shouldn't be going anywhere near it."
"threading adds some additional automatic accounting, as well as several convenience utilities, all of which makes it the preferred option for standard Python code."

**Comment by younesr1** [comment]
do we still want `./movo_common/movo_description/meshes/sensors/collision/kinect_one_collision.stl`? its not used?

File: movo_desktop/movo_viz/launch/view_robot.launch (hunk 3 above)
Comment (younesr1) [reply movo_desktop/movo_viz/launch/view_robot.launch]: hmm why does move_group need to be launched with view_robot.launch? did u just try adding different launch files until it worked? i dont think move_group should be launched with view_robot

File: movo_robot/movo_bringup/launch/perception/movo_perception.launch [hunk 6]
```diff
@@ -1,112 +1,150 @@
...
+
+  <arg name="stereo_module/exposure/1"  default="7500"/>
+  <arg name="stereo_module/gain/1"      default="16"/>
+  <arg name="stereo_module/exposure/2"  default="1"/>
+  <arg name="stereo_module/gain/2"      default="16"/>
+  
+  
+
+  <group ns="$(arg camera)">
+    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
```
Comment (younesr1) [inline movo_robot/movo_bringup/launch/perception/movo_perception.launch:77]: does this launch the nodelet manager for u?

File: movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py (hunk 4 above)
Comment (FelixKHWong) [reply movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py]: Good catch I didn't know about that and I was hesitant to change the code but after reading more of the code the thread library is only used to create a mutex and acquire or release so I think I've correctly replaced the functionality with the new threading library(coming in the next push)

**Comment by FelixKHWong** [comment]
> do we still want `./movo_common/movo_description/meshes/sensors/collision/kinect_one_collisi`

I thought I deleted it before but I missed it. I'll remove it now

File: movo_desktop/movo_viz/launch/view_robot.launch (hunk 3 above)
Comment (FelixKHWong) [reply movo_desktop/movo_viz/launch/view_robot.launch]: view_robot_err
Yeah I tried to resolve the errors I found when running the launch file with roslaunch. I added the movo_upload to get the robot description and then the movo_group launch to fix a semantic description error. If this runs on master fine then perhaps I'm missing a change and I can remove the lines

File: movo_robot/movo_bringup/launch/perception/movo_perception.launch (hunk 6 above)
Comment (FelixKHWong) [reply movo_robot/movo_bringup/launch/perception/movo_perception.launch:77]: managernode
nodes
 I think it's opening a manager for the camera. I'm not that familiar with nodes so I'm not sure if there's supposed to be a node for everything or something

File: movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py (hunk 4 above)
Comment (FelixKHWong) [reply movo_common/movo_third_party/moveit_python/src/moveit_python/planning_scene_interface.py]: https://peps.python.org/pep-3108/#obsolete

**Comment by FelixKHWong** [comment]
Screenshot from 2022-10-15 13-36-46
I finally figured out how to sudo with no password for the ros file. I'll have to edit the sudoers file on movo so I'll do that next time I'm in

**Comment by FelixKHWong** [comment]
> looks good to me, only hesitation is it seems like there were significantly more files removed than added. I'm assuming this is just the kinect and realsesne being structured differently?

Yeah most of the kinect stuff is inside the movo repo but for realsense we include it as a ROS package so the code for it is in /opt

**Review by younesr1 — APPROVED** [review]
lgtm! good stuff felix!

**Comment by younesr1** [comment]
actually theres one more thing i think u should do
- [x] update the Camera section in the dev_notes.txt

**Comment by younesr1** [comment]
so to be clear, none of these changes will be ported the robohub fork of this repo, right? your 2 commit names imply this

**Comment by FelixKHWong** [comment]
> so to be clear, none of these changes will be ported the robohub fork of this repo, right? your 2 commit names imply this

Yeah this is only intended for those who want to use a realsense with the movo. This commit should not be taken for those who are using the default movo setup

**Pull request merged**